x/gov/client/cli/query.go

```diff
@@ -24,7 +29,905 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			// validate that the proposal id is a uint
+			proposalID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("proposal-id %s not a valid int, please input a valid proposal-id", args[0])
+			}
+
+			// check to see if the proposal is in the store
+			proposalRes, err := queryClient.Proposal(
+				context.Background(),
+				&types.QueryProposalRequest{ProposalId: proposalID},
+			)
+			if err != nil {
+				return fmt.Errorf("failed to fetch proposal-id %d: %s", proposalID, err)
+			}
+
+			propStatus := proposalRes.GetProposal().Status
+			if !(propStatus == types.StatusVotingPeriod || propStatus == types.StatusDepositPeriod) {
+				resByTxQuery, err := QueryVotesByTxQuery(clientCtx, types.QueryProposalRequest{ProposalId: proposalID})
+				if err != nil {
+					return err
+				}
+
+				var votes types.Votes
+				// TODO migrate to use JSONMarshaler (implement MarshalJSONArray
+				// or wrap lists of proto.Message in some other message)
+				clientCtx.LegacyAmino.MustUnmarshalJSON(resByTxQuery, &votes)
+				return clientCtx.PrintOutput(votes)
+
+			}
+
+			res, err := queryClient.Votes(
+				context.Background(),
+				&types.QueryVotesRequest{ProposalId: proposalID},
+			)
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintOutput(res)
+
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdQueryDeposit implements the query proposal deposit command. Command to
+// get a specific Deposit Information
+func GetCmdQueryDeposit() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deposit [proposal-id] [depositer-addr]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Query details of a deposit",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query details for a single proposal deposit on a proposal by its identifier.
+Example:
+$ %s query gov deposit 1 1
+`,
+				version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			// validate that the proposal id is a uint
+			proposalID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("proposal-id %s not a valid uint, please input a valid proposal-id", args[0])
+			}
+
+			// check to see if the proposal is in the store
+			_, err = queryClient.Proposal(
+				context.Background(),
+				&types.QueryProposalRequest{ProposalId: proposalID},
+			)
+			if err != nil {
+				return fmt.Errorf("failed to fetch proposal-id %d: %s", proposalID, err)
+			}
+
+			depositorID, _ := cmd.Flags().GetUint64(flagDepositor)
```

Comment:
I think this should be depositerAddr

Reply:
Also, should we use arg[1] or flagDepositer here?

Reply:
it is id only